plugin/src/main/kotlin/com/automattic/android/applystyle/ApplyStylePlugin.kt

```diff
@@ -0,0 +1,167 @@
+/*
+ * This Kotlin source file was generated by the Gradle 'init' task.
+ */
+package com.automattic.android.applystyle
+
+import org.gradle.api.Project
+import org.gradle.api.Plugin
+
+import io.gitlab.arturbosch.detekt.DetektPlugin
+import io.gitlab.arturbosch.detekt.extensions.DetektExtension
+import org.gradle.api.plugins.quality.CodeQualityExtension
+import java.io.File
+import java.io.IOException
+import java.nio.file.Path
+
+/**
+ * A custom plugin that standardizes the styles across various projects.
+ *
+ * For now the plugin deals only with the configuration of Detekt.
+ */
+class ApplyStylePlugin: Plugin<Project> {
+    override fun apply(project: Project) {
+        project.tasks.register("extractDetektConf") { task ->
+            task.doLast {
+                val detektExtension = project.getDetektExtension() as DetektExtension
+                val file = detektExtension.config.singleFile ?: path(project.buildDir,"detekt.yml")
```

Comment:
Wouldn't `detektExtension.config.singleFile` result in a `IllegalStateException` if the project has multiple detekt configuration files?

---

Is `build/detekt.yml` a valid location for Detekt? I can't find it in their documentation and considering `build` folder is supposed to be a throwaway folder since `clean` task will literary do that, is it a good location to download a configuration file?

If we wanted a task such as this, and I am not entirely convinced about that to be honest but will talk about that separately, wouldn't it be better to accept a parameter for the file location so it can be selected when the task is ran?